### ISSUE TITLE
[Databricks dialect] PARTITIONED BY and CLUSTER BY clauses are unparsable for CREATE MATERIALIZED VIEW for Lakeflow Declarative Pipelines

### ISSUE THREAD

**Issue by KikeSenpai** [sqlfluff/sqlfluff#7135]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While linting queries similar to these:

```sql
-- With PARTITIONED BY clause
create or refresh materialized view my_simple_mv (
    col1 string comment 'Dummy column 1',
    col2 int comment 'Dummy column 2',
    col3 boolean comment 'Dummy column 3'
)
partitioned by (col1)
comment 'Example simplified materialized view with dummy fields.'
tblproperties ('quality' = 'gold')
as select
    col1,
    col2,
    col3
from my_source_table;

-- With CLUSTER BY clause
create or refresh materialized view my_simple_mv (
    col1 string comment 'Dummy column 1',
    col2 int comment 'Dummy column 2',
    col3 boolean comment 'Dummy column 3'
)
cluster by (col1)
comment 'Example simplified materialized view with dummy fields.'
tblproperties ('quality' = 'gold')
as select
    col1,
    col2,
    col3
from my_source_table;
```

### Expected Behaviour

To lint the sql files without error

### Observed Behaviour

The following errors were shown:
```bash
== [dummy_queries/dummy_query_mv_2.sql] FAIL
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: "cluster
                       | by (col1)\ncomment 'Example simpl..."
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
== [dummy_queries/dummy_query_mv.sql] FAIL
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section:
                       | "partitioned by (col1)\ncomment 'Example s..."
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
```

### How to reproduce

Create SQL files with the examples queries I gave in this issue and run sqlfluff lint, we are using uv to manage packages so I put them in a folder and ran:

```bash
uv run sqlfluff lint ./dummy_queries/*.sql
```

### Dialect

Databricks

### Version

3.4.2

### Configuration

```toml
[tool.sqlfluff.core]
exclude_rules = "RF03, RF01, ST05"
templater = "placeholder"
runaway_limit = 10
max_line_length = 120
indent_unit = "space"
dialect = "databricks"

[tool.sqlfluff.indentation]
tab_space_size = 4
indented_on_contents = false
allow_implicit_indents = true

[tool.sqlfluff.layout.type.comma]
spacing_before = "touch"
line_position = "trailing"

[tool.sqlfluff.rules.aliasing.table]
aliasing = "explicit"

[tool.sqlfluff.rules.aliasing.column]
aliasing = "explicit"

[tool.sqlfluff.rules.aliasing.expression]
allow_scalar = false

[tool.sqlfluff.rules.aliasing.length]
min_alias_length = 2

[tool.sqlfluff.rules.ambiguous.column_references]
group_by_and_order_by_style = "implicit"

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.identifiers]
capitalisation_policy = "lower"
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.functions]
capitalisation_policy = "lower"
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.literals]
capitalisation_policy = "lower"

[tool.sqlfluff.rules.capitalisation.types]
capitalisation_policy = "lower"
extended_capitalisation_policy = "lower"

[tool.sqlfluff.rules.convention.not_equal]
preferred_not_equal_style = "c_style"

[tool.sqlfluff.rules.convention.terminator]
multiline_newline = false

[tool.sqlfluff.templater.placeholder]
param_style = "dollar"
source_catalog = "parloa-prod-weu"
target_catalog = "parloa-prod-weu"
target_schema = "live"
job_date = "2025-01-01"
lookback_days = 1
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KikeSenpai** [comment]
Could someone support in my PR?
the linting error doesn't seem to be related to my changes
